code/messaging/build.gradle.kts

```diff
@@ -50,6 +51,7 @@ aepLibrary {
 dependencies {
     implementation("com.adobe.marketing.mobile:core:$mavenCoreVersion")
     compileOnly("com.google.firebase:firebase-messaging:23.4.1")
+    implementation("org.jetbrains.kotlinx:kotlinx-serialization-json:1.5.1")
```

Comment:
Let us do this conversion manually for now. as there is only one template, and then discuss with how the SDK team feels about using Kotlinx serialization.

Reply:
ContentCardProvider can do the parsing instead of SmallImageTemplate so that the solution extension knows about the specific keys instead of the UI library